lte/gateway/c/oai/tasks/amf/amf_authentication.h

```diff
@@ -0,0 +1,62 @@
+/**
+ * Copyright 2020 The Magma Authors.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+#include <sstream>
+#include <thread>
+namespace magma5g {
+
+// Authentication related procedure
+typedef struct nas5g_amf_auth_proc_s {
+  nas_amf_common_proc_t amf_com_proc;
+  nas5g_timer_t T3560; /* Authentication timer         */
+  unsigned int retransmission_count;
+  amf_ue_ngap_id_t ue_id;
+  bool is_cause_is_registered;  //  could also be done by seeking parent
+                                //  procedure
+  ksi_t ksi;
+  uint8_t rand[AUTH_RAND_SIZE]; /* Random challenge number  */
+  uint8_t autn[AUTH_AUTN_SIZE]; /* Authentication token     */
+  int amf_cause;
+} nas5g_amf_auth_proc_t;
+
+typedef struct nas5g_auth_info_proc_s {
+  nas5g_cn_proc_t cn_proc;
+  bool request_sent;
+  amf_ue_ngap_id_t ue_id;
+  bool resync;  // Indicates whether the authentication information is requested
+                // due to sync failure
+} nas5g_auth_info_proc_t;
+
+nas5g_auth_info_proc_t* nas_new_5gcn_auth_info_procedure(
```

Comment:
No need for the `new` qualifier, there is only one nas5g auth info procedure
```suggestion
nas5g_auth_info_proc_t* nas5g_cn_auth_info_procedure(
```

Reply:
fixed